File: server/java/src/main/java/com/hrms/business/abstracts/EmployeeService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.Employee;

import java.util.List;

public interface EmployeeService {
    Result save(Employee employee);
    DataResult<List<Employee>> getAll();
    DataResult<Employee> getById(int id);
}

File: client/react/hrms/src/services/employeeService.js
import axios from "axios";

export default class EmployeeService {
  getAll() {
    return axios.get("http://localhost:8080/api/employees/getall");
  }
  getById(id) {
    return axios.get("http://localhost:8080/api/employees/getbyid?id=" + id);
  }
  save(employer) {
    return axios.post("http://localhost:8080/api/employees/save", employer);
  }
}
